app/src/main/java/org/schabi/newpipe/settings/NewPipeSettings.java

```diff
@@ -110,7 +111,9 @@ private static String getNewPipeChildFolderPathForDir(final File dir) {
     }
 
     public static boolean useStorageAccessFramework(final Context context) {
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        if (DeviceUtils.isFireTv()) {
+            return false;
+        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
             return true;
         } else if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
             return false;
```

Comment:
Can the 2 `return false` cases be combined with `||`?

Reply:
Done.

Reply:
There should be a comment for each condition. I forgot to request that when reviewing the SAF PR.
```suggestion
        if (DeviceUtils.isFireTv()
            || Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
             // SAF file open/close dialogues can't be confirmed on FireTV (see #6455)
             // SAF is not supported on Android versions prior Lollipop
            return false;
        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
            return true;
```
@Stypox I forgot the reason for using SAF on Android 10 and above only.

Reply:
Edit: delete me, I am stupid and late to the game :laughing:

Reply:
So is this fine as-is, or do I add a comment for Lollipop? I think the Lollipop part is fairly explanatory, while I added a comment for the FireTV part because it's not immediately obvious.

Reply:
All good.